### ISSUE TITLE
How to calculate p?

### ISSUE THREAD

**Issue by BlueBlueFF** [Alibaba-MIIL/ASL#18]
In paper, "For very hard negative samples (with p > p∗, where p∗ is defined as the point where d (dp/dL)/dz=0,"
How to calculate p∗?

**Comment by BlueBlueFF** [comment]
And any the ablation study on resnet？

**Comment by mrT23** [comment]
there is no analytical solution to this equation
it can easily be solved numerically for given hyper-parameters

in practice, combining focusing with a small probability margin (m=0.05< p∗) worked well for us on all datasets.

regarding resnet, will be released in V2 of the paper in 2-3 weeks

**Comment by BlueBlueFF** [comment]
> there is no analytical solution to this equation
> it can easily be solved numerically for given hyper-parameters
> 
> in practice, combining focusing with a small probability margin (m=0.05< p∗) worked well for us on all datasets.
> 
> regarding resnet, will be released in V2 of the paper in 2-3 weeks

Thanks for your reply, And one more question: how to set weight( like in BCEWithLogitsLoss) in ASL loss?

**Comment by mrT23** [comment]
ASL doesn't have a "weight" hyperparameter and does not need it.
